addons/purchase/migrations/9.0.1.2/end-migration.py

```diff
@@ -1,20 +1,21 @@
 # -*- coding: utf-8 -*-
 # Copyright 2019 Tecnativa - Jairo Llopis
+# Copyright 2019 Tecnativa - Pedro M. Baeza
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
 
 from openupgradelib import openupgrade
 
 
 @openupgrade.logging()
 def deferred_pol_qty_received_compute(env):
-    """Complete migration of purchase.order.line#qty_received
-
-    Skip records that were optimized in SQL in post-migration.
+    """Complete migration of purchase.order.line#qty_received for products
+    with BoMs. We don't go further in initial filtering, as this number should
+    be low and it's not going to be heavy to be computed.
     """
-    # Execute https://bit.ly/2PgUKaT for matching records
+    if 'mrp.bom' not in env:
+        return  # mrp not installed
     pol_todo = env["purchase.order.line"].search([
-        ("order_id.state", "in", ['purchase', 'done']),
-        ("product_id.type", "in", ['consu', 'product']),
```

Comment:
You shouldn't delete these lines; they are there to exclude records that were already computed in post migration, and AFAICS that are gonna hit these lines without need if bom is installed: https://github.com/OCA/OpenUpgrade/blob/195b61a948dff1c64279eb68f054433b34006b51/addons/purchase/purchase.py#L493-L498

Reply:
Included again